public/utils/message_content_pool.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject, Observable, Subscriber } from 'rxjs';
+
+const JOB_INTERVAL = 50;
+const CONTENT_SLICE_LENGTH = 10;
```

Comment:
Should we make the actual streaming speed configurable?

Reply:
Yeah, I was trying to but can not find a use case for that. Updated anyway, dev for the future.